tests/baselines/reference/exportDefaultNamespace.js

```diff
@@ -0,0 +1,32 @@
+//// [tests/cases/compiler/exportDefaultNamespace.ts] ////
+
+//// [a.ts]
+// https://github.com/Microsoft/TypeScript/issues/3792
+export default namespace N {
+    export function foo(): number {
+        return 0xC0FFEE;
+    }
+ }
+
+//// [b.ts]
+import A from './a';
+
+const x = A.foo();
+
+//// [a.js]
+"use strict";
+exports.__esModule = true;
+// https://github.com/Microsoft/TypeScript/issues/3792
+var N = exports.N || {};

```

Comment:
exports.N is a bit awkward here, but AFAIK should always be undefined, as things like the following aren't allowed?
```
export default class N {}
export class N {}
```

Reply:
Thanks for the feedback, I'll get started on it. Would a nameless namespace be equivalent to a "namespace expression"? In any case I see some issues when module=es6:

```ts
export default namespace {
    export const a = 2;
}
// namespace merge
export default namespace {
    export const b = 2;
}

// requires generated variable name:
var _gen0 = {}
(function (_gen0) {
    _gen0.a = 2;
})(_gen0);
(function (_gen0) {
    _gen0.b = 2;
})(_gen0);
```

As anonymous interfaces aren't allowed either currently, maybe this would be better handled as a separate issue?

Reply:
That's a fair point, and perhaps we should hold off on nameless enum/namespace for now until the team has had an opportunity to discuss. My intuition is that you would not be able to merge with a nameless default namespace.

@DanielRosenwasser, @mhegazy: any thoughts?